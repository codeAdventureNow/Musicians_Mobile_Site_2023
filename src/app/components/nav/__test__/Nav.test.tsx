import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import mockRouter from 'next-router-mock';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';
import Nav from '../nav';

const user = userEvent.setup();

describe('Nav', () => {
  it('should render the nav bar', () => {
    const { asFragment } = render(<Nav />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('tests that when a user click the logo it will route to home page', async () => {
    render(<Nav />, { wrapper: MemoryRouterProvider });
    user.click(screen.getByTestId('nav-logo'));
    await waitFor(() => expect(mockRouter.asPath).toEqual('/'));
  });

  it('tests that "About" link clicked routes to the "about" page', async () => {
    render(<Nav />, { wrapper: MemoryRouterProvider });
    user.click(screen.getByTestId('About-id'));
    await waitFor(() => expect(mockRouter.asPath).toEqual('/about'));
  });

  it('tests that when a user click the "schedule a free lesson" button it will route to "contact" page', async () => {
    render(<Nav />, {
      wrapper: MemoryRouterProvider,
    });
    user.click(screen.getByTestId('schedule-button'));
    await waitFor(() => expect(mockRouter.asPath).toEqual('/contact'));
  });
});
