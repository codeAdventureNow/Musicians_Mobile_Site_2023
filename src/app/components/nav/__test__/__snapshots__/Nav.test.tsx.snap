// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render the nav bar 1`] = `
<DocumentFragment>
  <header
    class="header"
    data-testid="navbar"
  >
    <div
      class="logo"
    >
      <a
        data-testid="nav-logo"
        href="/"
      >
        <img
          alt="musicians mobile logo"
          data-nimg="1"
          decoding="async"
          height="100"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2FLogo.jpg&w=256&q=75"
          srcset="/_next/image?url=%2Fimages%2FLogo.jpg&w=128&q=75 1x, /_next/image?url=%2Fimages%2FLogo.jpg&w=256&q=75 2x"
          style="color: transparent; max-width: 100%; height: auto;"
          width="100"
        />
      </a>
    </div>
    <ul
      class="navbar"
    >
      <a
        class="link"
        data-testid="Home-id"
        href="/"
      >
        Home
      </a>
      <a
        class="link"
        data-testid="About-id"
        href="/about"
      >
        About
      </a>
      <a
        class="link"
        data-testid="Clients-id"
        href="/clients"
      >
        Clients
      </a>
      <a
        class="link"
        data-testid="Teachers-id"
        href="/teachers"
      >
        Teachers
      </a>
      <a
        class="link"
        data-testid="FAQ-id"
        href="/faq"
      >
        FAQ
      </a>
      <a
        class="link"
        data-testid="Contact-id"
        href="/contact"
      >
        Contact
      </a>
    </ul>
    <div
      class="cta_burgerMenu"
    >
      <div
        class="nav_cta"
      >
        <a
          class="cta_button"
          data-testid="schedule-button"
          href="/contact"
          rel="noreferrer"
        >
          Schedule A Free Lesson!
        </a>
        <br />
        <a
          class="phone"
          href="tel:408-960-1100"
        >
           
          <img
            alt="phone_icon"
            class="phone_icon"
            data-nimg="1"
            decoding="async"
            height="100"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Fphone_icon.png&w=48&q=75"
            srcset="/_next/image?url=%2Fimages%2Fphone_icon.png&w=32&q=75 1x, /_next/image?url=%2Fimages%2Fphone_icon.png&w=48&q=75 2x"
            style="color: transparent; max-width: 100%; height: auto;"
            width="20"
          />
          <p>
            (408) 960-1100
          </p>
        </a>
      </div>
      <div
        class="nav-icon1 "
        id="nav_icon1"
      >
        <span />
        <span />
        <span />
      </div>
    </div>
  </header>
</DocumentFragment>
`;
